R/CageData_Integration.R

```diff
@@ -0,0 +1,191 @@
+#' Converts different type of files to Granges
+#' Only Accepts bed files for now
+#' @param x A file containing column-based counts
+#' @param bed6 If bed6, no meta column is added
+toGR=function(x, bed6 = TRUE){
+  if(!bed6){
+    gr<- GRanges(x[, 1],IRanges(x[, 2]-1, x[, 3]))
+    return(gr)
+  }
+  starts<- ifelse(x[, 6]=="+",x[, 2]-1, x[, 2])
+  ends<- ifelse(x[, 6]=="-",x[, 3], x[, 3]-1)
+  gr<- GRanges(x[, 1],IRanges(starts,ends))
+  strand(gr)<-x[, 6]
+  score(gr)<-x[, 5]
+  if(ncol(x) > 6)  mcols(gr)=x[, 7:ncol(x)]
+  return(gr)
+}
+
+
+#' Get CageData as granges
+#' @param dataName The location of the cage-file
+#' @param filterValue The number of counts(score) to filter on for a tss to pass as hit
+#' @importFrom data.table fread
+getFilteredCageData = function(dataName, filterValue = 1){
+  if (.Platform$OS.type == "unix") {
+    if (file.exists(dataName)) {
+      if(gsub(pattern = ".*\\.","",dataName) == "gzip"){
+      rawCageData <- toGR(as.data.frame(fread(paste("gunzip -c",dataName), sep = "\t")))
+      }else if(gsub(pattern = ".*\\.","",dataName) == "bed"){
+        rawCageData <- toGR(as.data.frame(fread(dataName, sep = "\t")))
+      }else{stop("Only bed and gzip formats are supported for filePath")}
+    }else{stop("Filepath specified does not name existing file.") }
+  }else{stop("Only unix operating-systems currently support filePath,use cage as GRanges argument instead.") }
+  if(is.null(rawCageData$score)) stop("Found no score column in the cageData-file, bed standard is column 5")
+  message("Loaded cage-file successfully")
+  filteredrawCageData <- rawCageData[rawCageData$score > filterValue,] #filter on score
+
+  #chromosome nameing, need to find a smart way for this, it wont work for all
+  seqlevels(filteredrawCageData) <- sub("chrY","Y",seqlevels(filteredrawCageData))
+  seqlevels(filteredrawCageData) <- sub("chrX","X",seqlevels(filteredrawCageData))
+  seqlevels(filteredrawCageData) <- sub("chrM","MT",seqlevels(filteredrawCageData))
+  return(filteredrawCageData)
+}
+
+#' When reassigning Transcript start site, often you want to add downstream too.
+#' This is a simple way to do that
+#' @param fiveUTRs The 5' leader sequences as GRangesList
+#' @param cds If you want to extend 5' leaders downstream, to catch uorfs going into cds, include it.
+addFirstCdsOnLeaderEnds = function(fiveUTRs, cds){
+
+  if(length(cds) == 0){
+    message("cds is empty, returning without using it.")
+    return(fiveUTRs)
+  }
+  if(is.null(names(cds))){
+    message("cds have no names, returning without using it.")
+    return(fiveUTRs)
+  }
+
+  if(tryCatch({length(cds[names(fiveUTRs)]) == 0},error = function(err) return(T))){ # <- check valid names
```

Comment:
that is the weirdest thing ever, remove trycatch, do names(cds), names(fiveUTR) with %in% or with setdiff